crates/trade/src/cfd.rs

```diff
@@ -104,9 +105,24 @@ pub fn calculate_pnl(
     Ok(pnl)
 }
 
+/// The closing price that the trader gets based on bid and ask price
+///
+/// The closing price is always for the trade because the trader requests to close a position.
+/// If the trader closes a long position then the trader gets the best bid price because the trader
+/// is effectively going short to close a long order.
+/// If the trader closes a short position then the trader gets the best ask price because the trader
+/// is effectively going long to close a short order.
+pub fn closing_price(trader_direction: Direction, quote: Quote) -> Decimal {
```

Comment:
What do you think of implementing this on `Quote`?

Reply:
I would then not call it `closing_price` but `price_for_direction` or so.

Reply:
It's hard to fit this into my mental model, but I can see how it could make the code easier :)

For me the `Quote` does not care about any underlying financial instrument, it just depicts the best bid and ask price in the market (weighted by volume).
For me "closing" is always related to the position, not to the quote - the quote does not care about closing.
Given that the quote defines the closing price I can see how it could somehow make sense to add it there, but if I would just look at the quote out of context I would be confused.

Reply:
Maybe you can explain your mental model - it might help to make all this code better :)

Reply:
>  For me "closing" is always related to the position, not to the quote - the quote does not care about closing.

Exactly, that's why I want to get rid of the term `closing` :D 

I would add something like: 

```
impl Quote {
    fn get_price_for_direction(&self, direction: Direction) -> Decimal {
        match direction {
            Direction::Long => self.ask_price,
            Direction::Short => self.bid_price,
        }
    }
}
```

and then from the outside, i.e. where you currently call `closing_price` you do: 
```
get_price_for_direction(position.direction.oposite())`
```

Then we know, we want the price for the opposite direction.